src/zjs_uart.c

```diff
@@ -161,21 +161,23 @@ static jerry_value_t uart_write(const jerry_value_t function_obj,
 
     zjs_make_promise(promise, NULL, NULL);
 
-    if (!jerry_value_is_object(argv[0])) {
-        DBG_PRINT("first parameter must be a Buffer");
+    uint32_t len;
+    void* data = zjs_serialize_data(argv[0], NULL, &len);
```

Comment:
Do you intend to be changing this API so that strings and byte arrays are now valid, because you seem to be doing that. If so, I guess update the API?